src/autobot/internal/steps/abdiffdrawstep.cpp

```diff
@@ -0,0 +1,36 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "abdiffdrawstep.h"
+
+#include "log.h"
+#include "../draw/abpaintprovider.h"
+
+#include "libmscore/draw/drawtypes.h"
+#include "libmscore/draw/drawcomp.h"
+
+using namespace mu::autobot;
+
+void AbDiffDrawStep::doRun(AbContext ctx)
+{
+    draw::Diff diff = ctx.val<draw::Diff>(AbContext::Key::DiffDrawData);
+    AbPaintProvider::instance()->setDiff(diff);
+    AbPaintProvider::instance()->setIsDiffDrawEnabled(true);
+
+    dispatcher()->dispatch("dev-notationview-redraw");
```

Comment:
Can you please explain the purpose of this?

Reply:
I need to call the notation redrawing from the test.
I thought how best to do this, variants:
1. Add view notations to the global context - but there can be several of them at the same time (one in the test)
2. We can do redrawn notation view indirectly, change something in the notation itself, or change the zoom and return it back, something like that - but I don't like this variant.
3. Add dev actions - I selected this variant.